business/istio_config.go

```diff
@@ -987,8 +987,49 @@ func (in *IstioConfigService) IsGatewayAPI(cluster string) bool {
 	return in.userClients[cluster].IsGatewayAPI()
 }
 
-func (in *IstioConfigService) GatewayAPIClasses(cluster string) []config.GatewayAPIClass {
-	return kubernetes.GatewayAPIClasses(in.IsAmbientEnabled(cluster))
+// API classes can come from three different places depending on the configuration:
+// 1. From explicitly listed classes in the configuration
+// 2. From classes matching a label selector in the configuration
+// 3. If neither are configured, all classes that use Istio as a controller
+func (in *IstioConfigService) GatewayAPIClasses(cluster string) ([]config.GatewayAPIClass, error) {
```

Comment:
This can now return an error, which doesn't match the other functions for this service. If preferred, I can silently ignore/drop errors instead, but this wouldn't be ideal for users.